frontend/src/core/codemirror/language/utils.ts

```diff
@@ -69,3 +69,18 @@ export function splitEditor(editor: EditorView) {
     afterCursorCode,
   };
 }
+
+export function extractHighlightedCode(editor: EditorView) {
+  const code = editor.state.doc.toString();
+  let { from, to } = editor.state.selection.main;
```

Comment:
you could for `if (from === to) {return ["", code]`

Reply:
or could do `if (from === to) {return null` to signify nothing was highlighted. it will force consumers of this downstream to check `null`.

i think that is better / clearer

Reply:
Implemented this and it seems to have fixed the stale issue